### PR TITLE
Fix Linter errors blocking CI

### PR DESCRIPTION
**Pull request by alex-hall**
This is mostly refactoring the test suite to use async/await instead of promises. 

The only exception to this was the section where Q promises were being intentionally tested. I figured most people using those style promises aren't also using async/await in conjunction, so I added an eslint ignore there. 

I can re-write those using async/await if there is interest. 

Also, the catch block in the runOrRetry method didn't actually do anything, so I took the liberty of throwing that exception instead of returning an array nowhere.

### CODE REVIEW

**Comment by alex-hall** [comment]
Also, once this PR is accepted, we need to update dependencies. 

I have done this locally and all tests still pass FYI..

File: test/job.js [hunk 1]
```diff
@@ -787,15 +796,14 @@ describe('Job', () => {
       agenda.on('complete', () => {
         if (!finished && processed.length === 3) {
           finished = true;
-          done();
         }
       });
 
       agenda.start();
 
-      agenda.schedule(new Date(now + 100), 'blocking', {i: 1}).then(() => {});
```
[the remark this answers is not in the document]
Comment (alex-hall) [reply test/job.js]: Ah.. that's a good point. I can fix that.

File: test/job.js [hunk 2]
```diff
@@ -249,22 +249,26 @@ describe('Job', () => {
...
+        const result = await mongoDb
+          .collection('agendaJobs')
+          .find({
+            _id: job.attrs._id
+          })
+          .toArray();
+
+        expect(result).to.have.length(0);
+      } catch (e) {
+        throw e;
```
[the remark this answers is not in the document]
Comment (alex-hall) [reply test/job.js]: Gotcha. I've run into issues with async tests failing silently with some jest tests, so it's a habit, but I can change it

File: lib/utils/process-jobs.js [hunk 3]
```diff
@@ -285,9 +285,12 @@ module.exports = function(extraJob) {
...
           // CALL THE ACTUAL METHOD TO PROCESS THE JOB!!!
           debug('[%s:%s] processing job', job.attrs.name, job.attrs._id);
-          job.run()
-            .catch(err => [err, job])
-            .then(job => processJobResult(...Array.isArray(job) ? job : [null, job]));
+          try {
+            const runJob = await job.run();
+            processJobResult(...Array.isArray(runJob) ? runJob : [null, runJob]);
+          } catch (err) {
+            throw err;
```
Comment (wingsbob) [inline lib/utils/process-jobs.js]: This is not the same behaviour.

```javascript
try {
    processJobResult(null, await job.run());
  } catch (err) {
    processJobResult(err);
  }
```

is probably closer, but then you potentially call `processJobResult` again if it is what throws the error 🤔

**Comment by alex-hall** [comment]
@wingsbob all comments should be fixed by now.

**Pull request merged**